utils/ExportUtils.js

```diff
@@ -0,0 +1,154 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2018 Extremely Heavy Industries Inc.
+ */
+
+import {XH} from '@xh/hoist/core';
+import {ToastManager} from '@xh/hoist/toast';
+import {fmtDate} from '@xh/hoist/format';
+import {orderBy, uniq, isString, isFunction} from 'lodash';
+import download from 'downloadjs';
+
+
+/**
+ * Cell formats used in excel exports.
+ * Can be applied to columns via the 'exportFormat' prop.
+ */
+export const exportFormats = {
+    DEFAULT:            'General',
+
+    // Numbers
+    NUM:                '0',
+    NUM_1DP:            '0.0',
+    NUM_2DP:            '0.00',
+    NUM_4DP:            '0.0000',
+
+    NUM_DELIMITED:      '#,##0',
+    NUM_DELIMITED_1DP:  '#,##0.0',
+    NUM_DELIMITED_2DP:  '#,##0.00',
+    NUM_DELIMITED_4DP:  '#,##0.0000',
+
+    LEDGER:             '#,##0_);(#,##0)',
+    LEDGER_1DP:         '#,##0.0_);(#,##0.0)',
+    LEDGER_2DP:         '#,##0.00_);(#,##0.00)',
+    LEDGER_4DP:         '#,##0.0000_);(#,##0.0000)',
+
+    LEDGER_COLOR:       '#,##0_);[Red](#,##0)',
+    LEDGER_COLOR_1DP:   '#,##0.0_);[Red](#,##0.0)',
+    LEDGER_COLOR_2DP:   '#,##0.00_);[Red](#,##0.00)',
+    LEDGER_COLOR_4DP:   '#,##0.0000_);[Red](#,##0.0000)',
+
+    // Rounding suffixes - these can be appended to the patterns above to round numbers
+    THOUSANDS:          ',"k"',
+    MILLIONS:           ',,"m"',
+    BILLIONS:           ',,,"b"',
+
+    // Percent
+    PCT:                '0%',
+    PCT_1DP:            '0.0%',
+    PCT_2DP:            '0.00%',
+    PCT_4DP:            '0.0000%',
+
+    // Dates
+    DATE_FMT:           'yyyy-mm-dd',
+    DATETIME_FMT:       'yyyy-mm-dd h:mm AM/PM'
+};
+
+/**
+ * Exports a grid to either Excel or CSV.
+ *
+ * Columns can install the following properties to manage how they are exported:
+ *
+ *      {string} exportName - specified title to appear at top of grid
+ *      {string|function} exportValue - modifies the value used in export
+ *              If string, can be used to point to a different field on the record
+ *              If function, can be used to transform the value
```

Comment:
Need to add docs for parameters to this function.  Was momentarily confused by the docs on the columns (which are also useful.).  

Perhaps this whole thing could be made in to a class 'ExportManager' with a single public static method  exportToFile.  Might give you a bit more room to document everything around export in a class comment.